src/nameserver/name_server_impl.cc

```diff
@@ -2127,10 +2127,17 @@ void NameServerImpl::MakeSnapshotNS(RpcController* controller, const MakeSnapsho
         return;
     }
     if (request->offset() > 0) {
-        thread_pool_.AddTask(boost::bind(&NameServerImpl::MakeTablePartitionSnapshot, this, request->pid(),
-                                         request->offset(), table_info));
-        response->set_code(::openmldb::base::ReturnCode::kOk);
-        return;
+        if (table_info->storage_mode() != common::kMemory) {
```

Comment:
this logic is ok. `DiskTableSnapshot` does not support snapshot with end_offset > 0